lib/rgb_matrix/animation.ex

```diff
@@ -1,109 +1,88 @@
-        }
-  defstruct [:type, :tick, :speed, :delay_ms, :loop, :next_frame, :frames]
-
-  @type animation_type ::
+  @type type ::
           __MODULE__.CycleAll
-          | __MODULE__.CycleLeftToRight
+          | __MODULE__.HueWave
           | __MODULE__.Pinwheel
-          | __MODULE__.Static
+          | __MODULE__.RandomSolid
+          | __MODULE__.RandomKeypresses
+          | __MODULE__.SolidColor
+          | __MODULE__.Breathing
+          | __MODULE__.SolidReactive
 
   @doc """
   Returns a list of the available types of animations.
   """
-  @spec types :: list(animation_type)
+  @spec types :: list(type)
   def types do
     [
       __MODULE__.CycleAll,
-      __MODULE__.CycleLeftToRight,
-      __MODULE__.Pinwheel
+      __MODULE__.HueWave,
+      __MODULE__.Pinwheel,
+      __MODULE__.RandomSolid,
+      __MODULE__.RandomKeypresses,
+      __MODULE__.SolidColor,
+      __MODULE__.Breathing,
+      __MODULE__.SolidReactive
     ]
   end
 
-  @type animation_opt ::
-          {:type, animation_type}
-          | {:frames, list}
-          | {:tick, non_neg_integer}
-          | {:speed, non_neg_integer}
-          | {:delay_ms, non_neg_integer}
-          | {:loop, non_neg_integer | :infinite}
-
-  @spec new(opts :: list(animation_opt)) :: Animation.t()
-  def new(opts) do
-    animation_type = Keyword.fetch!(opts, :type)
-    frames = Keyword.get(opts, :frames, [])
+  @doc """
+  Returns an animation's initial state.
+  """
+  @spec new(animation_type :: type, leds :: list(LED.t())) :: {render_in, t}
+  def new(animation_type, leds) do
+    config_module = Module.concat([animation_type, Config])
+    animation_config = config_module.new()
+    {render_in, animation_state} = animation_type.new(leds, animation_config)
 
-    %Animation{
+    animation = %__MODULE__{
       type: animation_type,
-      tick: opts[:tick] || 0,
-      speed: opts[:speed] || 100,
-      delay_ms: opts[:delay_ms] || 17,
-      loop: opts[:loop] || :infinite,
-      frames: frames,
-      next_frame: List.first(frames)
+      config: animation_config,
+      state: animation_state
     }
-  end
 
-  @doc """
-  Updates the state of an animation with the next tick of animation.
-  """
-  @spec next_frame(animation :: Animation.t()) :: Animation.t()
-  def next_frame(animation) do
-    next_frame = animation.type.next_frame(animation)
-    %Animation{animation | next_frame: next_frame, tick: animation.tick + 1}
+    {render_in, animation}
   end
 
   @doc """
-  Returns the frame count of a given animation,
-
-  Note: this function returns :infinite for dynamic animations.
+  Returns the next state of an animation based on its current state.
   """
-  @spec frame_count(animation :: Animation.t()) :: non_neg_integer | :infinite
-  def frame_count(%{loop: :infinite}), do: :infinite
+  @spec render(animation :: t) :: {list(RGBMatrix.any_color_model()), render_in, t}
+  def render(animation) do
+    {colors, render_in, animation_state} =
```

Comment:
The functions above and below both return tuples which start with `render_in`. It would be nice for consistency to have this one do the same.
How about this:
```suggestion
    {render_in, colors, animation_state} =
```